orttraining/tools/onnx2tfevents/onnx2tfevents.py

```diff
@@ -342,12 +342,14 @@ def transform(self, graph: GraphProto) -> None:
 
 class AppendOpTypeTransformer(ReplaceNameTransformer):
     """Tensorboard's node search tool can only index the node name, not the node type. To make the op type searchable,
-    append the op type to the end of the node name.
+    add the op type to the beginning of the node name. From the tensorboard, we can also easily get the op type
+    even part of the node name is truncated.
     """
 
     def _generate_new_names(self, graph: GraphProto) -> None:
         for node in graph.node:
-            self.new_names[node.output[0]] = node.output[0] + "::" + node.op_type
+            pos = node.output[0].rfind("/")
+            self.new_names[node.output[0]] = node.output[0][: pos + 1] + node.op_type + "::" + node.output[0][pos + 1 :]
```

Comment:
```python
prefix, postfix = node.output[0].rsplit('/', 1)
name = f"{prefix}/{node.op_type}::{postfix}"
```

maybe consider the above? I would avoid array indexing in general as they are error prone. 

Just an example. If the existence of / is not guaranteed I should checks too

Reply:
The existence of / is not guaranteed, if not there, x,y="abc".rsplit("/",1) will get ValueError because the returned list has only one element, need extra if-else to handle. I think array indexing is OK as long as using it carefully. The code can handle both existant of / or not, which is clear.

Reply:
SG